### PR TITLE
task 2

### PR DESCRIPTION
**Pull request by YauheniLoban**

### CODE REVIEW

File: HomeTask2/HomeTask2/Program.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            string stringEnteredFromConsole;
+            do
+            {
+                Console.WriteLine("Enter a non-empty string");
+                stringEnteredFromConsole = Console.ReadLine();
+            }
+            while (stringEnteredFromConsole.Length == 0);
+            
+            SubstringOfCharactersWithEvenIndices EntryPointObject = new SubstringOfCharactersWithEvenIndices();
```
Comment (IgorKheidorov) [inline HomeTask2/HomeTask2/Program.cs]: code style

File: HomeTask2/HomeTask2/Program.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            {
+                Console.WriteLine("Enter a non-empty string");
+                stringEnteredFromConsole = Console.ReadLine();
+            }
+            while (stringEnteredFromConsole.Length == 0);
+            
+            SubstringOfCharactersWithEvenIndices EntryPointObject = new SubstringOfCharactersWithEvenIndices();         
+            Console.WriteLine(EntryPointObject.CreateSubstringOfEvenElements(stringEnteredFromConsole));
+        }
+
```
Comment (IgorKheidorov) [inline HomeTask2/HomeTask2/Program.cs]: Empty line

File: HomeTask2/HomeTask2/Program.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        /// <summary> 
+        /// Method CreateSubstringOfEvenElements
+        /// Function to create a string consisting of even elements of the transmitted string
+        /// </summary> 
+        /// <param name="stringForSelectingSubstring">String, which was inputed</param> 
+        /// <returns>Created substring</returns>
+        string CreateSubstringOfEvenElements(string stringForSelectingSubstring)
+        {
+            string returnSubstring;
```
Comment (IgorKheidorov) [inline HomeTask2/HomeTask2/Program.cs]: no need to create returnSubstring here, unnecessary extra object, bad name, better resultSubstring

File: HomeTask2/HomeTask2/Program.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        static void Main(string[] args)
+        {
+            string stringEnteredFromConsole;
+            do
+            {
+                Console.WriteLine("Enter a non-empty string");
+                stringEnteredFromConsole = Console.ReadLine();
+            }
+            while (stringEnteredFromConsole.Length == 0);
+
```
Comment (IgorKheidorov) [inline HomeTask2/HomeTask2/Program.cs]: ??? Why do you create class object inside method of the same class?

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]

File: HomeTask2/HomeTask2/Program.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+{
+    /// <summary> 
+    /// EntryPoint
+    /// Gets the string from console and prints a substring of elements with even indices to the console
+    /// </summary> 
+    class EntryPoint
+    {
+        static void Main(string[] args)
+        {
+            string stringEnteredFromConsole;
```
Comment (AlexanderSoroka) [inline HomeTask2/HomeTask2/Program.cs]: stupid name

File: HomeTask2/HomeTask2/SubstringOfCharactersWithEvenIndices.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
+﻿using System.Text;
+
+namespace DevTask2
+{
+    /// <summary> 
+    /// SubstringOfCharactersWithEvenIndices
+    /// Сreates a substring of elements with even indices to the console
+    /// </summary> 
+    class SubstringOfCharactersWithEvenIndices
```
Comment (AlexanderSoroka) [inline HomeTask2/HomeTask2/SubstringOfCharactersWithEvenIndices.cs]: Unreadable class name

File: HomeTask2/HomeTask2/SubstringOfCharactersWithEvenIndices.cs [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        /// </summary> 
+        /// <param name="stringForSelectingSubstring">String, which was inputed</param> 
+        /// <returns>Created substring</returns>
+        public string CreateSubstringOfEvenElements(string stringForSelectingSubstring)
+        {
+            StringBuilder allocatedSubstring = new StringBuilder();
+
+            for (int i = 0; i < stringForSelectingSubstring.Length; i += 2)
+            {
+                allocatedSubstring.Append(stringForSelectingSubstring[i]); // Add an element to the selected row
```
Comment (AlexanderSoroka) [inline HomeTask2/HomeTask2/SubstringOfCharactersWithEvenIndices.cs]: useless and incorrect comment

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

**Review by AlexanderSoroka — APPROVED** [review]

**Pull request merged**